server/src/main/java/org/jboss/as/server/services/net/LocalDestinationOutboundSocketBindingAddHandler.java

```diff
@@ -71,6 +79,30 @@ protected void performRuntime(final OperationContext context, final ModelNode op
 
         final PathAddress address = PathAddress.pathAddress(operation.get(OP_ADDR));
         final String outboundSocketName = address.getLastElement().getValue();
+
+        context.addStep(new OperationStepHandler() {
+            @Override
+            public void execute(OperationContext context, ModelNode operation) throws OperationFailedException {
+                final String currentSocketBindingGroupName = address.getParent().getLastElement().getValue();
+                final Set<ResourceEntry> children = context.readResourceFromRoot(PathAddress.EMPTY_ADDRESS)
+                        .getChildren(ModelDescriptionConstants.SOCKET_BINDING_GROUP);
+                for (Resource.ResourceEntry re : children) {
+                    if (re.getName().equalsIgnoreCase(currentSocketBindingGroupName)) {
```

Comment:
1) of 2)

Do you know if this is an issue with a managed domain config? There may be some validation for this problem already. 

If not, then we need to deal with the fact that socket-binding-group A can include B and C, and B can include D. Names need to be unique across the entire included set. I believe we check for this, although that check may need to be enhanced to deal with the fact that the unique names need to across socket-binding AND {remote|local}-destination-outbound-socket-binding.

Ideally the existing managed domain checks will handle this in the domain case, and then all this logic can be guarded by a check of context.getProcessType(), confirming that 1) processType.isServer() and 2) processType != ProcessType.DOMAIN_SERVER.